src/main/java/org/mqttbee/mqtt/handler/MqttChannelInitializer.java

```diff
@@ -17,31 +17,55 @@
 
 package org.mqttbee.mqtt.handler;
 
+import io.netty.channel.Channel;
 import io.netty.channel.ChannelInitializer;
 import io.netty.channel.ChannelPipeline;
-import io.netty.channel.socket.SocketChannel;
+import io.netty.handler.codec.http.HttpClientCodec;
+import io.netty.handler.codec.http.HttpObjectAggregator;
+import io.netty.handler.ssl.SslHandler;
 import io.reactivex.SingleEmitter;
+import io.reactivex.exceptions.Exceptions;
 import org.mqttbee.annotations.NotNull;
+import org.mqttbee.api.mqtt.MqttClientSslData;
 import org.mqttbee.api.mqtt.mqtt5.message.connect.connack.Mqtt5ConnAck;
 import org.mqttbee.mqtt.MqttClientData;
 import org.mqttbee.mqtt.codec.encoder.MqttEncoder;
+import org.mqttbee.mqtt.handler.websocket.MqttWebSocketClientProtocolHandler;
+import org.mqttbee.mqtt.handler.websocket.WebSocketBinaryFrameDecoder;
+import org.mqttbee.mqtt.handler.websocket.WebSocketBinaryFrameEncoder;
 import org.mqttbee.mqtt.message.connect.MqttConnect;
 import org.mqttbee.mqtt.handler.auth.MqttAuthHandler;
 import org.mqttbee.mqtt.handler.auth.MqttDisconnectOnAuthHandler;
 import org.mqttbee.mqtt.handler.connect.MqttConnectHandler;
 import org.mqttbee.mqtt.handler.disconnect.MqttDisconnectHandler;
 import org.mqttbee.mqtt.ioc.ChannelComponent;
 
+import javax.net.ssl.SSLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Optional;
+
+import static org.mqttbee.mqtt.datatypes.MqttVariableByteInteger.MAXIMUM_PACKET_SIZE_LIMIT;
+import static org.mqttbee.mqtt.handler.ssl.SslUtil.createSslHandler;
+
 /**
  * Default channel initializer.
  *
  * @author Silvio Giebl
+ * @author David Katz
  */
-public class MqttChannelInitializer extends ChannelInitializer<SocketChannel> {
+public class MqttChannelInitializer extends ChannelInitializer<Channel> {
```

Comment:
The methods of this class should be reordered

Reply:
done